easybuild/easyconfigs/d/DGL/DGL-1.1.3-foss-2022a-CUDA-11.7.0.eb

```diff
@@ -0,0 +1,143 @@
+# updated to version 1.1.3, based on DGL-0.9.1-foss-2021a-CUDA-11.3.1
+# GKlib-METIS added as module as third-party approach does not build
+# libxsmm set to 'off' so it is using the EasyBuild module
+
+easyblock = 'CMakeMake'
+
+name = 'DGL'
+version = '1.1.3'
+versionsuffix = '-CUDA-%(cudaver)s'
+
+homepage = 'https://www.dgl.ai'
+description = """DGL is an easy-to-use, high performance and scalable Python package for deep learning on graphs.
+DGL is framework agnostic, meaning if a deep graph model is a component of an end-to-end application, the rest
+of the logics can be implemented in any major frameworks, such as PyTorch, Apache MXNet or TensorFlow."""
+
+toolchain = {'name': 'foss', 'version': '2022a'}
+
+github_account = 'dmlc'
+source_urls = [GITHUB_LOWER_SOURCE]
+sources = [
+    {
+        'download_filename': 'v%(version)s.tar.gz',
+        'filename': '%(namelower)s-%(version)s.tar.gz',
+    },
+    {
+        'source_urls': ['https://github.com/KarypisLab/METIS/archive'],
+        'download_filename': 'v5.2.1.tar.gz',
+        'filename': 'metis-5.2.1.tar.gz',
+        'extract_cmd': "tar -C %(namelower)s-%(version)s/third_party/METIS --strip-components=1 -xf %s",
```

Comment:
Hm, similar to the rest of the third party things, we do have METIS, nanoflann and such, and i don't think anyone is stopping us from adding a CCCL and the rest as well.. so i'm not sure why these were kept in as sources? @akesandgren comment?

Reply:
Yes, I agree, we got already existing EC for them. However, I could not get that working with external builds. Also, it appears that they nailed it down to specific commits as well. So in the end I decided to fall back to that approach but I am happy to get that working with existing EC if somebody can show me how to do that without unpicking everything.

Reply:
The METIS they use is patched in some incompatible way.
The EC i made does have nanoflann as a dependency, but I forgot to comment it out in the "use_externals_instead_of_submodules" patch.

And since they patch METIS I didn't even consider using an external GKlib-METIS for that reason.

Reply:
I could try and see if `nanoflann` is still working. At one point, to be honest, I decided to make it working so it is reproducible instead of spending too much time flogging what looked like a dead horse to me.